html/browsers/history/joint-session-history/joint-session-history-iframe-state.html

```diff
@@ -0,0 +1,40 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>Joint session history should not override parent's state.</title>
+<meta name="timeout" content="long">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<iframe id="frame" src="about:blank"></iframe>
+<script>
+async_test(function(t) {
+    t.step_timeout(() => {
+        // Setup.
+        var initialState = {foo: 'bar'};
+        var newStateFromChild = {foo: 'baz'};
+        var child = document.getElementById("frame");
+
+        // Perform navigation in the top-level container to set the state.
+        window.history.pushState(initialState, 'title');
+
+        // Validate initial state was properly set.
+        assert_object_equals(window.history.state, initialState);
+
+        child.onload = t.step_func_done(() => {
+            // Child's initial state should be `null`.
+            assert_equals(child.contentWindow.history.state, null);
+
+            // Navigate in the child.
+            child.contentWindow.history.pushState(newStateFromChild, 'title');
+
+            // Child's state should now equal the new state.
+            assert_object_equals(child.contentWindow.history.state, newStateFromChild);
+            // Parent's state should still be preserved, having exactly the same state as it
+            // had before parent navigated.
+            assert_object_equals(window.history.state, initialState);
+        })
+        child.src = "joint-session-history-filler.html";
+    }, 1000);
```

Comment:
Why do you have this timeout wrapper?

Reply:
I was looking at https://github.com/web-platform-tests/wpt/blob/d8189dabae6687709322ac251710f8a0444b1e34/html/browsers/history/joint-session-history/joint-session-history-remove-iframe.html for the example test that's similar to what I wanted to add. I suspect that this was due to the fact that we want to finish the navigation in the page before running the test.

TBH this isn't something that I like, since timeouts are never a good solution, so open to any changes that'd work here 👍

Reply:
Do you notice a difference when running it without this wrapper? It should all be equivalent as far as I can tell.

Reply:
You are right - seems like it's actually not needed. Will 🔥